### ISSUE TITLE
Accessibility - Item Behind another

### ISSUE THREAD

**Issue by felire** [facebook/react-native#29557]
## Description

 If I have two items overlapped, for example a fixed header and an item behind, if I touch on the header with VoiceOver or TalkBack it automatically select the item behind instead of the header, I tried with `zIndex` style prop or with `importantForAccessibility` prop (android) and it's still not working

I resolved on android, wrapping the View that I want to give priority on a `TouchableOpacity`. That solution isn't work on ios. 
## React Native version:
Last

## Steps To Reproduce
You should have two siblings View where one of these Views should be behind the other. Then you should give accessibility props to the View at the front. And you will see that when you click on the front View, VoiceOver will read the data of the View at the back.

## Expected Results
It should read the Accessibility Props of the View at the front

There isn't option of disabling Accessibility props for the others Views

**Comment by felire** [comment]
I'm using a SectionList to render the header

**Comment by rectified95** [comment]
Please, add a more concrete repro, such as a Snack or at least a code snippet. We also require environment info before accepting issues.

**Comment by hqdai** [comment]
I can the same issue, this is my example project [Interactive.zip](https://github.com/facebook/react-native/files/6538112/Interactive.zip) for re-produce steps. It seems the header section is always behind the item section. So we can't touch the header section when scrolling down. :(

<img width="950" alt="Screen Shot 2021-05-25 at 16 20 17" src="https://user-images.githubusercontent.com/19548213/119473530-74015600-bd75-11eb-916d-b3bffb4c25dd.png">
